### PR TITLE
Deprecated functions

### PR DESCRIPTION
**Pull request by 00875**

### CODE REVIEW

File: gamemode/modules/fpp/pp/client/menu.lua [hunk 1]
```diff
@@ -1076,7 +1076,7 @@ function FPP.PrivateSettings(Panel)
 	for k,v in pairs(PrivateSettings) do
 		local box = vgui.Create("DCheckBoxLabel")
 		box:SetText("I don't want to "..k)
-		box:SetValue(tobool(GetConVarNumber("FPP_PrivateSettings_"..v)))
+		box:SetValue(GetConVar("FPP_PrivateSettings_"..v):GetBool())
```
Comment (00875) [inline gamemode/modules/fpp/pp/client/menu.lua]: I edited FPP in DarkRP repository haha

File: entities/entities/fadmin_motd/cl_init.lua [hunk 2]
```diff
@@ -12,7 +12,7 @@ local function WorldToScreen(vWorldPos,vPos,vScale,aRot)
 end
 
 function ENT:LoadPage()
-	local Page = GetConVarString("_FAdmin_MOTDPage")
+	local Page = GetConVar("_FAdmin_MOTDPage"):GetString()
```
Comment (Bo98) [inline entities/entities/fadmin_motd/cl_init.lua]: There is much better ways to do this. There is a reason why GetConVarString is deprecated.

File: gamemode/modules/fadmin/fadmin/logging/sv_init.lua [hunk 3]
```diff
@@ -16,7 +16,7 @@ end
 local LogFile
 function FAdmin.Log(text, preventServerLog)
 	if not text or text == "" then return end
-	if not tobool(GetConVarNumber("FAdmin_logging")) then return end
+	if !GetConVar("FAdmin_logging"):GetBool() then return end
```
Comment (Bo98) [inline gamemode/modules/fadmin/fadmin/logging/sv_init.lua]: Sneaky change of coding style.

**Comment by Bo98** [comment]
Pretty sure you missed some deprecated functions as I remember changing more functions in my copy.

File: gamemode/modules/fadmin/fadmin/access/sv_init.lua [hunk 4]
```diff
@@ -356,5 +356,5 @@ FAdmin.StartHooks["Access"] = function() --Run all functions that depend on othe
 	FAdmin.Commands.AddCommand("immunity", toggleImmunity)
 	FAdmin.Commands.AddCommand("SetImmunity", setImmunity)
 
-	FAdmin.SetGlobalSetting("Immunity", (GetConVarNumber("_FAdmin_immunity") == 1 and true) or false)
+	FAdmin.SetGlobalSetting("Immunity", (GetConVar("_FAdmin_immunity"):GetBool()) or false)
```
Comment (Bo98) [inline gamemode/modules/fadmin/fadmin/access/sv_init.lua]: I don't think you need the `or false` here.

File: gamemode/modules/afk/cl_afk.lua [hunk 5]
```diff
@@ -1,4 +1,4 @@
-local TextColor = Color(GetConVarNumber("Healthforeground1"), GetConVarNumber("Healthforeground2"), GetConVarNumber("Healthforeground3"), GetConVarNumber("Healthforeground4"))
+local TextColor = Color(GetConVar("Healthforeground1"):GetFloat(), GetConVar("Healthforeground2"):GetFloat(), GetConVar("Healthforeground3"):GetFloat(), GetConVar("Healthforeground4"):GetFloat())
```
Comment (Bo98) [inline gamemode/modules/afk/cl_afk.lua]: Does `Color` work with floats? Might do but I'm not sure if it does.

File: gamemode/modules/fadmin/fadmin/serversettings/cl_init.lua [hunk 6]
```diff
@@ -17,11 +17,11 @@ local function SetLimits()
...
 		if v.type == "Numeric" then
 			local left, right = Form:NumberWang(v.text, nil, v.low or 0, v.high or 1000, v.decimals or 0 )
-			left:SetFloatValue(GetConVarNumber(v.name))
-			left:SetValue(GetConVarNumber(v.name))
+			left:SetFloatValue(GetConVar(v.name):GetFloat())
+			left:SetValue(GetConVar(v.name):GetFloat())
 
 			function left:OnValueChanged(val)
-				if val == GetConVarNumber(v.name) then
+				if val == GetConVar(v.name):GetFloat() then --???
```
Comment (Bo98) [inline gamemode/modules/fadmin/fadmin/serversettings/cl_init.lua]: What's the `--???` about?

File: entities/entities/fadmin_motd/cl_init.lua (hunk 2 above)
Comment (Bo98) [inline entities/entities/fadmin_motd/cl_init.lua]: This expands to other changes in this pull request too (but not all).

File: gamemode/modules/afk/cl_afk.lua (hunk 5 above)
Comment (HDG-MrDeeJay) [inline gamemode/modules/afk/cl_afk.lua]: I'm pretty sure it does.

Comment (Bo98) [inline gamemode/modules/afk/cl_afk.lua]: I didn't check and based it on experience with RGB in other places (eg. Android uses integers).

File: gamemode/modules/fadmin/fadmin/serversettings/cl_init.lua (hunk 6 above)
Comment (00875) [inline gamemode/modules/fadmin/fadmin/serversettings/cl_init.lua]: I don't sure what is dis

File: gamemode/modules/afk/cl_afk.lua (hunk 5 above)
Comment (00875) [inline gamemode/modules/afk/cl_afk.lua]: 2015-08-28_15-40_garry s mod

File: entities/entities/fadmin_motd/cl_init.lua (hunk 2 above)
Comment (FPtje) [inline entities/entities/fadmin_motd/cl_init.lua]: I wouldn't be surprised if the definition of GetConvarString is literally the C variant of  `return GetConvar(x):GetString()`

**Comment by FPtje** [comment]
This PR contains syntax errors god damnit

**Comment by FPtje** [comment]
The inefficiencies of this PR can be fixed later. It will be a smaller change than from without this PR.

**Comment by Bo98** [comment]
Were you using the github.com editor or do you like thousands of commits?

**Pull request merged**